File: Data Structures & Algorithms/permutations-ii/submission-0.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        n=len(nums)
        used=[0]*n
        ans=[]
        def sets(temp,used):
            if len(temp)==n:
                ans.append(temp.copy())
                return

            for i in range(n):
                if used[i]:
                    continue
                temp.append(nums[i])
                used[i]=1
                sets(temp,used)
                used[i]=0
                temp.pop()
        sets([],used)

        return [list(x) for x in set(tuple(x) for x in ans)]
        
        

File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n=nums.size();
        vector<int> dp(n,0);
        dp[0]=1;
        for(int i=0;i<n;i++){
            if(!dp[i]){
                return false;
            }
            int temp=nums[i];
            int j=1;
            while(temp && i+j<n){
                dp[i+j]=1;
                j++;
                temp--;
            }
        }
        return dp[n-1]==1 ? true : false;
        
    }
};
